parley/src/layout/cursor.rs

```diff
@@ -852,8 +851,29 @@ impl Selection {
             }
             AnchorBase::Line(start, end) => {
                 let target = Self::line_from_point(layout, x, y);
-                let [anchor, focus] =
-                    cursor_min_max(layout, [target.anchor, target.focus, start, end]);
+                let [anchor, focus] = extend_selection(target, [start, end]);
+                Self {
+                    anchor,
+                    focus,
+                    anchor_base: self.anchor_base,
+                    h_pos: None,
+                }
+            }
+        }
+    }
+
+    /// Returns a new selection with the focus extended to the given point.
+    #[must_use]
+    pub fn shift_click_extension<B: Brush>(&self, layout: &Layout<B>, x: f32, y: f32) -> Self {
+        let target = Cursor::from_point(layout, x, y);
+        match self.anchor_base {
+            AnchorBase::Cluster => Self::new(self.anchor, target),
+            AnchorBase::Word(start, end) | AnchorBase::Line(start, end) => {
+                let [anchor, focus] = if target.index < start.index {
+                    [end, target]
+                } else {
+                    [start, target]
+                };
                 Self {
                     anchor,
                     focus,
```

Comment:
Thoughts on:
```suggestion
                let anchor = if target.index < start.index {
                    end
                } else {
                    start
                };
                Self {
                    anchor,
                    focus: target,
```
The amount of space taken up for such a simple condition is a bit unfortunate, but it keeps the code clear.